platform/x86/platform.mk

```diff
@@ -39,21 +39,31 @@ endif
 
 CFLAGS := $(CSFLAGS) -Wall -Wextra -Werror -std=gnu11 -Wno-discarded-qualifiers \
           -Wno-unused-variable -Wno-unused-parameter -Wsign-conversion -Wpointer-arith \
-          -ffunction-sections -fdata-sections \
+          -ffunction-sections -fdata-sections -pthread \
           $(ARCH_CFLAGS) $(addprefix -D,$(CDEFINES))
 
+ifeq (clang,$(COMPILER))
+  CFLAGS := $(filter-out -Wno-discarded-qualifiers,$(CFLAGS)) -Wno-missing-field-initializers \
+            -Wno-incompatible-pointer-types-discards-qualifiers -Wno-missing-braces
+endif
+
 # Linker flags
 LDFLAGS := -lrt
 
 # Platform targets
-.PHONY: run gdb
+.PHONY: run gdb socketcan
 
 run: $(BIN_DIR)/$(PROJECT)$(PLATFORM_EXT)
 	@$<
 
 gdb: $(GDB_TARGET)
 	@$(GDB) $<
 
+socketcan:
```

Comment:
I can't call this from the main `Makefile` is that intentional?

Reply:
Moved to `Makefile`